contrib/errorprone/src/python/pants/contrib/errorprone/tasks/errorprone.py

```diff
@@ -115,6 +117,12 @@ def calculate_sources(self, target):
     return {source for source in target.sources_relative_to_buildroot()
             if source.endswith(self._JAVA_SOURCE_EXTENSION)}
 
+  @memoized_property
+  def _javac_jar_path(self):
+    targets = list(self.context.resolve('//:errorprone'))
```

Comment:
Given that you've registered this tool, you should be able to get its classpath by calling `self.tool_classpath('errorprone')`, and then you can filter out entries from the classpath directly in that list?

Alternatively, having two tool registrations (one for errorprone, and one for errorprone's compiler plugin) would also be cleaner.

Reply:
`self.tool_classpath('errorprone')` only returns the shaded jar.

Adding a separate registration for errorprone-javac should be easier though.  I'll give that a try.

Reply:
@cheister : If you disable shading (ie, don't pass a main method) it will give you a list of classpath entries.

Reply:
Yeah, I wasn't sure if I wanted to disable shading just to get the errorprone-javac jar.  Adding errorprone-javac as it's own jvm_tool also means we don't have to include the IvyTaskMixin which seems better anyway.